cmd/amqpproxy/main_test.go

```diff
@@ -70,26 +70,35 @@ func TestAMQPProxy(t *testing.T) {
 func TestAMQPProxyExcludePayloadData(t *testing.T) {
 	testEnv.SkipIfNotLive(t)
 
-	testData := mustCreateAMQPProxyExcludePayloadData(t, []string{})
+	transformerOptions := []string{
+		"--exclude-payload-data",
+	}
+	testData := mustCreateAMQPProxy(t, transformerOptions)
 
-	receiver, err := testData.ServiceBusClient.NewReceiverForQueue(testData.ServiceBusQueue, nil)
+	sender, err := testData.ServiceBusClient.NewSender(testData.ServiceBusQueue, nil)
 	require.NoError(t, err)
 
 	defer func() {
-		err := receiver.Close(context.Background())
+		err := sender.Close(context.Background())
 		require.NoError(t, err)
 	}()
 
-	sender, err := testData.ServiceBusClient.NewSender(testData.ServiceBusQueue, nil)
+	err = sender.SendMessage(context.Background(), &azservicebus.Message{Body: []byte("hello world")}, nil)
+	require.NoError(t, err)
+
+	receiver, err := testData.ServiceBusClient.NewReceiverForQueue(testData.ServiceBusQueue, nil)
 	require.NoError(t, err)
 
 	defer func() {
-		err := sender.Close(context.Background())
+		err := receiver.Close(context.Background())
 		require.NoError(t, err)
 	}()
 
-	err = sender.SendMessage(context.Background(), &azservicebus.Message{Body: []byte("hello world")}, nil)
+	messages, err := receiver.ReceiveMessages(context.Background(), 1, nil)
 	require.NoError(t, err)
+	require.NotEmpty(t, messages)
```

Comment:
I can see checking that messages Body values are actually still there (ie, we didn't accidentally create a bug where we erase the payload for real)

Reply:
Updated to check that the body of the message sent is as expected